SmartDeviceLink/SDLResponseDispatcher.m

```diff
@@ -140,6 +143,43 @@ - (void)clear {
     _audioPassThruHandler = nil;
 }
 
+- (void)sdl_removeOldButtonHandlersFor:(SDLRPCRequest *)request {
+    for(SDLRPCCorrelationId *key in self.rpcRequestDictionary.allKeys) {
+        if ([self.rpcRequestDictionary[key] isKindOfClass:[SDLAlert class]] && [request isKindOfClass:[SDLAlert class]]) {
+            SDLAlert *oldAlert = (SDLAlert *)self.rpcRequestDictionary[key];
+            SDLAlert *newAlert = (SDLAlert *)request;
+            if(newAlert.softButtons.count != 0) {
```

Comment:
If a new alert or scrollable message, come in, we don't need to check if the new alert has soft buttons. Because only one can be active at a time, we know we can simply clear the old ones.

Please format your braces consistently with an additional space as well.